### PR TITLE
feat: semantic releases and automatic changelog.

### PR DESCRIPTION
**Pull request by gardar**
* Adds a CI workflow for automatically generating changelog using antsibull-changelog and antsichaut.
* Adds a CI workflow for automatically building/publishing collection as releases (will add galaxy publish later, when collection is "released").

Both the changelog and collection build/publish workflows use semantic versioning and conventional commits to avoid toil.

We need to manually publish one release that has a semantic version (currently the latest version is tagged as `latest`). After that the workflow can take care of the rest.
I didn't want to add complexity to the workflow by supporting a non-semantic version label that's only going to be used initially.

### CODE REVIEW

**Comment by gardar** [comment]
CC: @SuperQ

**Comment by gardar** [comment]
The CI failures on this PR are caused because some of the workflow actions are expecting to find a config on the default branch, so they should only be present on this initial PR.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,26 +25,24 @@ To have your code merged, see the expectations listed below.
...
-way similar to circle ci keyword [`[ci skip]`](https://docs.travis-ci.com/user/customizing-the-build#Skipping-a-build))
-to a commit with merge request. Available keywords are (square brackets are important!):
+We try to stick to semantic versioning and our releases are automated. Release is created by assigning [conventional commits](https://www.conventionalcommits.org/)
+to PR title. Available keywords are:
  
-* `[patch]`, `[fix]`, `[bugfix]` - for PATCH version release
-* `[minor]`, `[feature]`, `[feat]` - for MINOR version release
-* `[major]`, `[breaking change]` - for MAJOR version release
+* `[patch]`, `[fix]`, `[bug]`, `[bugfix]`, `[test]`, `[refactor]`, `[packaging]`, `[docs]` - for PATCH version release
+* `[minor]`, `[enhancement]`, `[feat]`, `[deprecated]`, `[removed]`, `[security]` - for MINOR version release
```
Comment (paulfantom) [inline CONTRIBUTING.md:39]:
```suggestion
* `[minor]`, `[enhancement]`, `[feature]`, `[deprecated]`, `[removed]`, `[security]` - for MINOR version release
```

**Review by paulfantom — APPROVED** [review]
LGTM, just one minor nit.

**Comment by gardar** [comment]
[779c4f4](https://github.com/prometheus-community/ansible/pull/26/commits/779c4f4abde61a6e73599c254292637a4cad5445) Disabled the auto publish for now in favor of a draft, we can re-enable it later once the collection reaches a production ready stage.

**Review by SuperQ — APPROVED** [review]

**Pull request merged**